sympy/matrices/matrices.py

```diff
@@ -26,6 +26,8 @@
 
 def _iszero(x):
     """Returns True if x is zero."""
+    if len(x.free_symbols) == 0:
```

Comment:
It's not really necessary to calculate the length if you are validating against `0`.